compiler/src/iree/compiler/DispatchCreation/test/insert_tensor_barriers.mlir

```diff
@@ -0,0 +1,173 @@
+// RUN: iree-opt --split-input-file --pass-pipeline="builtin.module(util.func(iree-dispatch-creation-insert-tensor-barriers))" %s | FileCheck %s
+
+util.func public @simple_linalg(%arg0: tensor<4x8xf32>) -> tensor<4x8xf32> {
+  %empty = tensor.empty() : tensor<4x8xf32>
+  %0 = linalg.generic {
+    indexing_maps = [affine_map<(d0, d1) -> (d0, d1)>,
+                     affine_map<(d0, d1) -> (d0, d1)>],
+    iterator_types = ["parallel", "parallel"]
+  } ins(%arg0 : tensor<4x8xf32>) outs(%empty : tensor<4x8xf32>) {
+  ^bb0(%in: f32, %out: f32):
+    linalg.yield %in : f32
+  } -> tensor<4x8xf32>
+  util.return %0 : tensor<4x8xf32>
+}
+// CHECK-LABEL: util.func public @simple_linalg
+//  CHECK-SAME:     %[[ARG0:[a-zA-Z0-9]+]]: tensor<4x8xf32>
```

Comment:
nit: the benefits of using regexp is that we can avoid matching the types. The types do not matter in the test, so I suggest removing the checks.